pkg/controller/daemon/daemon_controller.go

```diff
@@ -1071,7 +1071,7 @@ func (dsc *DaemonSetsController) syncNodes(ds *apps.DaemonSet, podsToDelete, nod
 		}
 		createWait.Wait()
 		// any skipped pods that we never attempted to start shouldn't be expected.
-		skippedPods := createDiff - batchSize
+		skippedPods := createDiff - (batchSize + pos)
```

Comment:
I gotta admit, I am super skeptical about this change (*and the existing code, too!*), and the test didn't convince me. :)

For things like this, instead of calculating a value after the fact, I think it is way safer to e.g. initialize a variable "remainingCreations" with the number of expected creations at the beginning of the loop, and then decrement it each time a creation is actually attempted (you might need to use sync/atomic functions for this). Then you don't ever have to worry about this math getting out of sync with the logic in the loop.

Reply:
I take quite a time to understand and figure out how to calculate the create expectations. And It does make sense to use `remainingCreations`. I'll keep working on it. :)